game/src/ui/gameplay/Arena/tile-selector/TileSelector.gd

```diff
@@ -23,38 +24,81 @@ func select(x : int, y : int) -> void:
 # if action == ACTION.BUILDING, type must be of Tower.TYPE
 # likewise if action == ACTION.DROPPING, type must be of Item.TYPE
 func set_action(action : int, type_resource_string : String) -> void:
+	print(action)
 	_action = action
 	_type = type_resource_string
 
 func _process(_delta: float) -> void:
-	self.modulate = Color(0, 1, 0) if _is_valid_tile() else Color(1, 0, 0)
-	
+	if (build_ui.tower_following_mouse == null):
+		self.modulate = Color(0, 1, 0) if _is_valid_tile() else Color(1, 0, 0)
+	else :
+		var mouse_tower = build_ui.tower_following_mouse
+		var mouse_pos = get_viewport().get_mouse_position()
+		mouse_tower.position = mouse_pos
+		mouse_tower.modulate = Color(0, 1, 0, 0.5) if _is_valid_tile() else Color(1, 0, 0, 0.5)
```

Comment:
I don't recommend this, when the modulation makes it impossible to tell if its blue or red: Maybe we could have very transparent / mostly opaque for invalid / valid placements? Or like modulate it black for invalid? idk just thinking

<img width="143" alt="image" src="https://user-images.githubusercontent.com/7360964/124405691-d9336880-dd71-11eb-9c38-6e0e3b13f338.png">
<img width="178" alt="image" src="https://user-images.githubusercontent.com/7360964/124405692-dafd2c00-dd71-11eb-8810-48dc18edd5e1.png">

Reply:
Another thing is that the tile (red / green wip logo) snaps to the 64x64 grid while the mouse moves, but the tile currently does not. Is there anyway to make this happen?

Reply:
resolved.

Reply:
nice, cleared both.